devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,550 @@
+private getCUSTOM_FAN_SPEED() { 0x0688 }
+private getCUSTOM_FAN_MODE() { 0x0698 }
+private getCUSTOM_THERMOSTAT_OPERATING_STATE() { 0x06BF }
+private getUNOCCUPIED_SETPOINT_CHILD_DEVICE_ID() {1}
+private getTHERMOSTAT_MODE_OFF() { 0x00 }
+private getTHERMOSTAT_MODE_AUTO() { 0x01 }
+private getTHERMOSTAT_MODE_COOL() { 0x02 }
+private getTHERMOSTAT_MODE_HEAT() { 0x03 }
+private getCUSTOM_FAN_MODE_ON() { 0x00 }
+private getCUSTOM_FAN_MODE_AUTO() { 0x01 }
+private getCUSTOM_FAN_MODE_CIRCULATE() { 0x02 }
+
+
+private getFAN_MODE_MAP() {
+	[
+			"00":"on",
+			"01":"auto",
+			"02":"circulate"
+	]
+}
+
+private getTHERMOSTAT_MODE_MAP() {
+	[
+			"00":"off",
+			"01":"auto",
+			"02":"cool",
+			"03":"heat",
+			"04":"heat"
+	]
+}
+
+private getTHERMOSTAT_OPERATING_STATE_MAP() {
+	[
+			"00":"idle",
+			"01":"cooling",
+			"02":"heating"
+	]
+}
+
+private getEFFECTIVE_OCCUPANCY_MAP() {
+	[
+			"00":"Occupied",
+			"01":"Unoccupied",
+			"02":"Override",
+			"03":"Standby"
+	]
+}
+
+def installed() {
+	log.debug "installed"
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	if (isViconicsVT8350()|| isSchneiderSE8350()) {
+		state.supportedFanModes = ["on", "auto"]
+	} else {
+		state.supportedFanModes = ["on", "auto", "circulate"]
+	}
+	state.supportedThermostatModes = ["off", "auto", "cool", "heat"]
+
+	sendEvent(name: "supportedThermostatFanModes", value: JsonOutput.toJson(state.supportedFanModes), displayed: false)
+	sendEvent(name: "supportedThermostatModes", value: JsonOutput.toJson(state.supportedThermostatModes), displayed: false)
+	sendEvent(name: "coolingSetpointRange", value: coolingSetpointRange, displayed: false)
+	sendEvent(name: "heatingSetpointRange", value: heatingSetpointRange, displayed: false)
+}
+
+private void createChildThermostat() {
+	log.debug "Creating child thermostat to handle unoccupied cooling/heating setpoints"
+	def label = "Unoccupied setpoints"
+	def childName = "${device.displayName} ${label}"
+
+	def child = addChildDevice("Child Thermostat Setpoints", "${device.deviceNetworkId}:1", device.hubId,
+			[completedSetup: true, label: childName, isComponent: true, componentName: "childSetpoints", componentLabel: label]
+	)
+
+	child.sendEvent(name: "coolingSetpoint", value: 20.0, unit: "C")
+	child.sendEvent(name: "heatingSetpoint", value: 21.0, unit: "C")
+	log.debug "child.inspect() ${child}"
+}
+
+def parse(String description) {
+	def eventMap = zigbee.getEvent(description)
+	def result = []
+
+	if (description?.startsWith("humidity: ")) {
+		// Viconics VT8350 humidity reports are parsed as floating point numbers (range 0 - 1%)
+		def humidityVal = (description - "humidity: " - "%").trim()
+		if (humidityVal.isNumber()) {
+			humidityVal = new BigDecimal(humidityVal) * 100
+		}
+		eventMap.name 	= "humidity"
+		eventMap.value 	= humidityVal
+		eventMap.unit 	= "%"
+	}
```

Comment:
Can you just handle this the way you're handling the other events? i.e. using the results of `parseDescriptionAsMap`?

Reply:
yes, it will be done as you suggest.